connaisseur/res/snapshot_schema.json

```diff
@@ -20,7 +20,7 @@
                 "meta": {
                     "type": "object",
                     "patternProperties": {
-                        "^(root|targets(\/\\w*)?)$": {
+                        "^(root|targets(\/[\\w_:\\.-]*)?)$": {
```

Comment:
is this the spec?

Reply:
According to the spec, it can be any string. https://github.com/theupdateframework/tuf/blob/f7695dace85444041489b83d5a66cd39c761bbd6/tuf/formats.py#L113

So i guess i'll change it to `"^(root|targets(\/.*)?)$"` ?

Reply:
We definitely should exempt some characters, like `/` (i.e. `[^/]`) or null bytes. I clearly vote for listing allowed characters explicitly and not use `.` though, otherwise you must be very certain to have thought of all characters to exempt.

Reply:
ok so back to `\w_:\.-` ? these are all that came to my mind

Reply:
I would say so. We could add `+`, `;`, `~` and the like, but I wouldn't assume anyone does that. What do others think? @annekebr @Starkteetje 

Also, `\w` includes `_`, and `.` does not need to be escaped when inside `[]`. (Inside a character class, each character stands for itself, except `\w` and the like, and `-` which denotes ranges unless it appears at the beginning or end of the character class.)

Reply:
AnyString allows whitespace, so we certainly dont want that

Doing `docker trust key generate something` I get

```
key name "hallo du" must start with lowercase alphanumeric characters and can include "-" or "_" after the first character
key name "hallo++du" must start with lowercase alphanumeric characters and can include "-" or "_" after the first character
key name "halloDu" must start with lowercase alphanumeric characters and can include "-" or "_" after the first character
```
so I guess, dash and underscore are sufficient in addition to lowercase alphanum